Core/GDCore/Extensions/Builtin/AdvancedExtension.cpp

```diff
@@ -77,7 +107,7 @@ void GD_CORE_API BuiltinExtensionsImplementer::ImplementsAdvancedExtension(
                     "",
                     "res/function32.png",
                     "res/function32.png")
-      .AddParameter("functionParameterName", "Parameter name")
+      .AddParameter("functionParameterName", _("Parameter name"), "number,string,boolean")
```

Comment:
Nitpicking: we could consider putting JSON inside the 3rd parameter. It's a bit more verbose but we've done it for some stuff already:
<img width="774" alt="image" src="https://user-images.githubusercontent.com/1280130/231701156-99a4c890-13e9-4fb4-95bf-18574c959e6f.png">

Up to you, might be overengineering.

Reply:
Event-functions can't declare "functionParameterName" parameters so it's never serialized. We can change it easily if we need to.